src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusExtensions.cs

```diff
@@ -124,19 +151,381 @@ public static IResourceBuilder<AzureServiceBusResource> AddQueue(this IResourceB
+                    topic.WriteJsonObjectProperties(writer);
+
+                    writer.WriteStartArray("Subscriptions");    //             "Subscriptions": [
+                    foreach (var subscription in topic.Subscriptions)
+                    {
+                        writer.WriteStartObject();              //               "{ (Subscription)"
+                        subscription.WriteJsonObjectProperties(writer);
+
+                        #region Rules
+                        writer.WriteStartArray("Rules");        //                 "Rules": [
+                        foreach (var rule in subscription.Rules)
+                        {
+                            writer.WriteStartObject();
+                            rule.WriteJsonObjectProperties(writer);
+                            writer.WriteEndObject();
+                        }
+
+                        writer.WriteEndArray();                 //                  ] (/Rules)
+                        #endregion Rules
+
+                        writer.WriteEndObject();                //               } (/Subscription)
+                    }
+
+                    writer.WriteEndArray();                     //             ] (/Subscriptions)
+
+                    writer.WriteEndObject();                    //           } (/Topic)
+                }
+                writer.WriteEndArray();                         //         ] (/Topics)
+
+                writer.WriteEndObject();                        //       } (/Namespace)
+                writer.WriteEndArray();                         //     ], (/Namespaces)
+                writer.WriteStartObject("Logging");             //     "Logging": {
+                writer.WriteString("Type", "File");             //       "Type": "File"
+                writer.WriteEndObject();                        //     } (/LoggingConfig)
+
+                writer.WriteEndObject();                        //   } (/UserConfig)
+                writer.WriteEndObject();                        // } (/Root)
+
+            }
+
+            // Apply ConfigJsonAnnotation modifications
+            foreach (var emulatorResource in serviceBusEmulatorResources)
+            {
+                var configFileMount = emulatorResource.Annotations.OfType<ContainerMountAnnotation>().LastOrDefault(v => v.Target == AzureServiceBusEmulatorResource.EmulatorConfigJsonPath);
+
+                // At this point there should be a mount for the Config.json file.
+                if (configFileMount == null)
+                {
+                    throw new InvalidOperationException("The configuration file mount is not set.");
+                }
+
+                var configJsonAnnotations = emulatorResource.Annotations.OfType<ConfigJsonAnnotation>();
+
+                foreach (var annotation in configJsonAnnotations)
+                {
+                    using var readStream = new FileStream(configFileMount.Source!, FileMode.Open, FileAccess.Read);
```

Comment:
Why write the whole file to disk above, only to read the file here and parse it? Can these 2 things happen at the same time?

Reply:
The file we are reading here could be the one we just populated, but it could be the one that was  provided as-is by the `WithConfigurationFile()` method. We still need to apply the custom callback from `ConfigureEmulator()`.